pkg/infra/game/agent/random.go

```diff
@@ -0,0 +1,23 @@
+package agent
+
+import (
+	"infra/game/decision"
+	"infra/game/state"
+	"math/rand"
+)
+
+type RandomAgent struct {
+}
+
+func (r RandomAgent) HandleFight(state state.State, baseAgent BaseAgent) {
+	fight := rand.Intn(1)%2 == 0
+	var action decision.FightAction
+	if fight {
```

Comment:
Maybe switch statements inside if and else block so that you actually fight when fight is true.

Reply:
Implemention detail up to the agent teams but from a style point of view: yes definitely.